Building with OpenCL-support fails on macOS with a case-sensitive file system

Hi!

I just noticed this issue while trying to build GDAL 3.5.1 from source using Homebrew on my Macbook with a case-sensitive APFS file system. It seems to also happen while trying to build from `master`.

Steps to reproduce:

```
cmake -S . -B build -DCMAKE_BUILD_TYPE=Release -DGDAL_USE_OPENCL=on
make -C build
```

Fails while building `gdalwarpkernel.cpp.o`:

```
[ 32%] Building CXX object alg/CMakeFiles/alg.dir/gdalwarpkernel.cpp.o
In file included from alg/gdalwarpkernel.cpp:59: alg/gdalwarpkernel_opencl.h:43:10: fatal error: 'OpenCL/OpenCL.h' file not found
#include <OpenCL/OpenCL.h>
         ^~~~~~~~~~~~~~~~~
alg/gdalwarpkernel_opencl.h:43:10: note: did not find header 'OpenCL.h' in framework 'OpenCL' (loaded from '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks')
1 error generated.
make[2]: *** [alg/CMakeFiles/alg.dir/gdalwarpkernel.cpp.o] Error 1
make[1]: *** [alg/CMakeFiles/alg.dir/all] Error 2
make: *** [all] Error 2
```

Seems like the header file is actually all lower-case `opencl.h`:

```
$ ls /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenCL.framework/Headers/
cl.h           cl_ext.h       cl_gl.h        cl_gl_ext.h    cl_platform.h  gcl.h          gcl_priv.h     opencl.h
```

I don't know if this only applies to the current macOS 12.4 SDK, but at least Apples (somewhat old) [documentation](https://developer.apple.com/library/archive/documentation/Performance/Conceptual/OpenCL_MacProgGuide/ExampleHelloWorld/Example_HelloWorld.html#//apple_ref/doc/uid/TP40008312-CH112-SW2) also includes it as `opencl.h` .

Patching the file allowed me to successfully build the project:

```diff
diff --git a/alg/gdalwarpkernel_opencl.h b/alg/gdalwarpkernel_opencl.h
index 9af64166fa..0bc2a209b9 100644
--- a/alg/gdalwarpkernel_opencl.h
+++ b/alg/gdalwarpkernel_opencl.h
@@ -40,7 +40,7 @@
 #define CL_TARGET_OPENCL_VERSION 100

 #ifdef __APPLE__
-#include <OpenCL/OpenCL.h>
+#include <OpenCL/opencl.h>
 #else
 #include <CL/opencl.h>
 #endif
```

Having a case-sensitive file system on macOS is of course not that common, most users use the default which is case-insensitive.
Feel free to close if this seems to be too niche of an issue. 🙂

--- a/alg/gdalwarpkernel_opencl.h
+++ b/alg/gdalwarpkernel_opencl.h
@@ -40,7 +40,7 @@
 #define CL_TARGET_OPENCL_VERSION 100
 
 #ifdef __APPLE__
-#include <OpenCL/OpenCL.h>
+#include <OpenCL/opencl.h>
 #else
 #include <CL/opencl.h>
 #endif

